dusty/log.py

```diff
@@ -57,3 +60,19 @@ def configure_client_logging():
     logging.basicConfig(stream=sys.stdout,
                         level=logging.INFO,
                         format='%(message)s')
+
+@contextlib.contextmanager
+def streaming_to_client():
+    """Puts the client logger into streaming mode, which sends
+    unbuffered input through to the socket one character at a time.
+    We also disable propagation so the root logger does not
+    receive many one-byte emissions. This context handler
+    was originally created for streaming Compose up's
+    terminal output through to the client and should only be
+    used for similarly complex circumstances."""
+    client_logger.propagate = False
+    handler = client_logger.handlers[0]
+    handler.append_newlines = False
+    yield
+    client_logger.propagate = True
+    handler.append_newlines = True
```

Comment:
this should really restore the old state not just clear the whole state to true right?